### PR TITLE
[TR] PIM-4639: fix cache clearer

### PR DESCRIPTION
**Pull request by sumbobyboys**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | ?
| Specs & Static     | Y
| Changelog updated  | N

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/sequential-edit.js [hunk 1]
```diff
@@ -170,7 +170,13 @@ define(
                 }, this));
             },
             followLink: function (event) {
-                this.goToProduct(event.currentTarget.dataset.id);
+                mediator.trigger('pim_enrich:form:state:confirm', {
+                    title: 'toto',
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/sequential-edit.js]: toto & tata ?

Comment (nidup) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/sequential-edit.js]: :couple:

File: features/Context/AssertionContext.php [hunk 2]
```diff
@@ -38,6 +38,20 @@ public function assertPageContainsText($text)
     }
 
     /**
+     * Checks, that page contains specified text.
```
Comment (phaseinducer) [inline features/Context/AssertionContext.php]: Wrong description

File: features/Context/AssertionContext.php [hunk 3]
```diff
@@ -38,6 +38,20 @@ public function assertPageContainsText($text)
...
+     *
+     * @Then /^(?:|I )should not see the text "(?P<text>(?:[^"]|\\")*)"$/
+     */
+    public function assertPageNotContainsText($text)
+    {
+        $this->getMainContext()->spin(function () use ($text) {
+            $this->assertSession()->pageTextNotContains($text);
+
+            return true;
+        }, 5);
```
Comment (phaseinducer) [inline features/Context/AssertionContext.php]: Does this really work ? If the text is here `pageTextNotContains()` throws an exception so I don't see the point spinning on it.

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/back-to-grid.js [hunk 4]
```diff
@@ -25,16 +26,16 @@ define(
...
+                this.$el.html(this.template({
+                    path: Routing.generate(
+                        'pim_enrich_product_index',
+                        {
+                            dataLocale: UserContext.get('catalogLocale')
+                        }
+                    )
+                }));
+
+                Navigation.getInstance().processClicks(this.$('a'));
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/back-to-grid.js]: What is that ? o_O

File: features/Context/Page/Base/Base.php [hunk 5]
```diff
@@ -31,6 +31,16 @@ class Base extends Page
...
 
     /**
+     * {@inheritdoc}
+     */
+    public function getElement($name, $timeout = 20, $message = 'no message')
+    {
+        return $this->spin(function () use ($name) {
+            return parent::getElement($name);
+        }, $timeout, $message);
+    }
```
Comment (jmleroux) [inline features/Context/Page/Base/Base.php]: could we not end by spin in another spin method ?

File: features/Context/Page/Product/Edit.php [hunk 6]
```diff
@@ -398,15 +402,13 @@ public function findFieldContainer($label)
             $label = explode(' in ', $label)[0];
         }
 
-        $labelNode = $this->find('css', sprintf('.field-container header label:contains("%s")', $label));
-        if (!$labelNode) {
-            throw new ElementNotFoundException($this->getSession(), 'label ', 'value', $label);
-        }
+        $labelNode = $this->spin(function() use ($label) {
+            return $this->find('css', sprintf('.field-container header label:contains("%s")', $label));
+        });
```
Comment (jmleroux) [inline features/Context/Page/Product/Edit.php]: no error message ?

File: features/Context/NavigationContext.php [hunk 7]
```diff
@@ -631,8 +633,13 @@ public function iShouldBeOnTheProductsPage()
      */
     public function iShouldBeOnTheProductEditPage(Product $product)
     {
-        $expectedAddress = $this->getPage('Product edit')->getUrl(['id' => $product->getId()]);
-        $this->assertAddress($expectedAddress);
+        $this->spin(function() use ($product) {
```
Comment (phaseinducer) [inline features/Context/NavigationContext.php]: Missing space

**Comment by phaseinducer** [comment]
GTM after fix

**Comment by jmleroux** [comment]
GTM

**Pull request merged**